### PR TITLE
Ipqualityscore module

### PR DESCRIPTION
**Pull request by krishnasism**

### CODE REVIEW

File: modules/sfp_ipqualityscore.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_ipqualityscore(SpiderFootPlugin):
+
+    meta = {
+        "name": "IPQualityScore",
+        "summary": "Determine if target is malicious using IPQualityScore API",
+        "flags": ["apikey"],
+        "useCases": ["Passive"],
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:24]: Should also have `Investigate`.

File: modules/sfp_ipqualityscore.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            error_str = ""
+        self.sf.error(f"Failed to get results for {qry}, code {res['code']}{error_str}")
+
+    def query(self, qry, eventName):
+        queryString = ""
+        if eventName == "PHONE_NUMBER":
+            queryString = f"https://ipqualityscore.com/api/json/phone/{self.opts['api_key']}/{qry}"
+        elif eventName == "EMAILADDR":
+            queryString = f"https://ipqualityscore.com/api/json/email/{self.opts['api_key']}/{qry}"
+        elif eventName == "IP_ADDRESS" or eventName == "DOMAIN_NAME":
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:102]: Does their `ip` endpoint accept domain names?

File: modules/sfp_ipqualityscore.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error(
+                f"You enabled {self.__class__.__name__} but did not set an API Key!"
+            )
+            self.errorState = True
+            return None
+
+        self.results[eventData] = True
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:144]: Even though you are storing this, you aren't doing anything with it. Was the intention to perform duplicate checking?

File: modules/sfp_ipqualityscore.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+        data = self.query(eventData, eventName)
+
+        if data is None:
+            return
+
+        fraudScore = data.get('fraud_score')
+        recentAbuse = data.get('recent_abuse')
+        botStatus = data.get('bot_status')
+        if fraudScore >= 85 or recentAbuse or botStatus:
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:154]: Please make the score a configurable option so the user can choose what their threshold is.

File: modules/sfp_ipqualityscore.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            evt = SpiderFootEvent("RAW_RIR_DATA", str(data), self.__name__, event)
+            self.notifyListeners(evt)
+
+            if eventName == "PHONE_NUMBER":
+                evt = SpiderFootEvent("MALICIOUS_PHONE_NUMBER", eventData, self.__name__, event)
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:159]: Please look at other modules producing `MALICIOUS_` events to see how they are formatted.

File: modules/sfp_ipqualityscore.py (hunk 2 above)
Comment (krishnasism) [reply modules/sfp_ipqualityscore.py:102]: Yes, I tested.

**Pull request merged**